windows-installer/stackdriver_logging_agent.nsi

```diff
@@ -0,0 +1,311 @@
+  
+  ; Register the software so it will appear in the add/remove programs menu
+  ${Print} "Registering ${DISPLAY_NAME}..."
+  ${RegisterUninstallSoftware} \
+    "${DISPLAY_NAME}" "${UNINSTALLER_LOCATAION}" "${UI_ICON}" "${COMPANY}" "$0"
+  
+  ; All done!
+  ${Print} "Instalation Complete"
+SectionEnd
+
+
+;--------------------------------
+; INSTALLER FUNCTIONS
+;--------------------------------
+
+; Function called as soon as the installer is initialized, before the GUI.
+Function .onInit
+  ; Set the install directory, we cannot do this at compile time as we are
+  ; using a runtime varaible.
+  ReadEnvStr $0 SYSTEMDRIVE 
+  StrCpy $INSTDIR "$0\${COMPRESSED_NAME}" 
+
+  ; Display the language selection dialog
+  !insertmacro MUI_LANGDLL_DISPLAY
+
+  ; Check for a previously installed installed version.  If one exists
+  ; prompt the user to remove it before continuing
+  ReadRegStr $0 SHCTX "${STACKDRIVER_LOGGING_UNINST_REG_KEY}" "UninstallString"
+  ${If} $0 != ""
+    ${RemoveOldVersion} "${DISPLAY_NAME}" $0
+  ${EndIf}
+  
+  ; Temporary directory used during install and automatically cleaned up after
+  InitPluginsDir
+FunctionEnd
+
+; Verify the install directory is correct.  There is a bug in an gem
+; that this install relies on.  If the file path has a space in it, it
+; will fail to install.  For now we disable the UI to change the path
+; but also need to disable it here in case of silent installs.
+Function .onVerifyInstDir
+  ; Check that the install direcotry has not changed.  If it has notify
+  ; the user and abort.
+  ReadEnvStr $0 SYSTEMDRIVE 
+  ${If} $INSTDIR != "$0\${COMPRESSED_NAME}"
+    MessageBox MB_OK "Invalid install directory '$INSTDIR', must be '$0\${COMPRESSED_NAME}'"
+    Abort
+  ${EndIf}
+FunctionEnd
+
+
+;--------------------------------
+; UNINSTALLER SECTIONS
+;--------------------------------
+
+Section "Uninstall"
+  ; Print messages to the details list view not the text (status) bar,
+  ; this provides a cleaner install when files are being removed.
+  SetDetailsPrint listonly
+  
+  ; Stop the fluentd service and show status
+  ;   '--reg-winsvc u'          -> Uninstall
+  ${UnPrint} "Stopping the ${DISPLAY_NAME}..."
+	${UnExecuteCommand} "${MAIN_INSTDIR}\bin\fluentd.bat" "--reg-winsvc u"
+	
+  ; Remove the software from the register
+	${UnPrint} "Unregistering the ${DISPLAY_NAME}..."
+	${RemoveRegisterUninstallSoftware} "${DISPLAY_NAME}"
+	
+  ; Clean up any other registry entires used
+	DeleteRegKey HKCU "${REG_KEY}"
+	
+	${UnPrint} "Cleaning up files in $INSTDIR..."
+
+  ; Delete all files in the main install directory.
+  RMDir /r /REBOOTOK ${MAIN_INSTDIR}
+  
+  ; Clean up all files not in the main directory
+  Delete /REBOOTOK ${UNINSTALLER_LOCATAION}
+  Delete /REBOOTOK ${FLUENTD_CONFIG_LOCATION}
+  Delete /REBOOTOK "$INSTDIR\${UI_ICON}"
+  
+  ; Remove the install directory if it is empty.
+	RMDir /REBOOTOK $INSTDIR
+  
+  ; All done!
+  ${UnPrint} "Uninstall Complete"
```

Comment:
The monitoring agent installer says "Uninstallation Complete" instead. Either is fine, but let's be consistent.